pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotConnection.java

```diff
@@ -67,15 +74,24 @@ public class PinotConnection extends AbstractBaseConnection {
     }
     _session = new org.apache.pinot.client.Connection(properties, brokers, transport);
 
-    _enableNullHandling = Boolean.parseBoolean(properties.getProperty(QueryOptionKey.ENABLE_NULL_HANDLING));
+    for (String possibleQueryOption: POSSIBLE_QUERY_OPTIONS) {
+      _queryOptions.put(possibleQueryOption, Boolean.parseBoolean(properties.getProperty(possibleQueryOption)));
+    }
   }
 
   public org.apache.pinot.client.Connection getSession() {
     return _session;
   }
 
-  public boolean isNullHandlingEnabled() {
-    return _enableNullHandling;
+  protected String enableQueryOptions(String sql) {
+    StringBuilder optionsBuilder = new StringBuilder();
+    for (HashMap.Entry<String, Boolean> optionEntry: _queryOptions.entrySet()) {
+      if (optionEntry.getValue() && !sql.contains(optionEntry.getKey())) {
+        optionsBuilder.append(DriverUtils.createSetQueryOptionString(optionEntry.getKey()));
```

Comment:
Should append a kv pair.

Reply:
Can you clarify? DriverUtils.createSetQueryOptionString does create a kv string.